### DISCUSSION TITLE
How to prevent saving new record by event handler?

### DISCUSSION THREAD

**Discussion by vovpff** [YetiForceCompany/YetiForceCRM#15565] [How to] [answered]
Dear Yeti crew
@mariuszkrzaczkowski @rskrzypczak 

Could you advice me how to prevent (abort) record saving process by event handler?

My users can't assume their thoughts and ideas in one comment message. I want to check if a comment was created within last 10 minutes and if it's true, update the last comment, and not save the current one at all. 

How to abort saving without creating record in database?

**Comment by mariuszkrzaczkowski** [comment, marked as answer]
Here is an example of a pre-save verification
https://github.com/YetiForceCompany/YetiForceCRM/blob/a90e0b74b1b791e3265bf982a8402dee1d33ee30/modules/OSSTimeControl/handlers/TimeControl.php#L83-L102

**Reply by vovpff** [reply]
@mariuszkrzaczkowski thank you.

What parameters available for response?

Is it possible to send result => false without message?

**Reply by mariuszkrzaczkowski** [reply]
https://github.com/YetiForceCompany/YetiForceCRM/blob/a90e0b74b1b791e3265bf982a8402dee1d33ee30/public_html/layouts/basic/modules/Vtiger/resources/Edit.js#L152-L165

**Reply by vovpff** [reply]
@mariuszkrzaczkowski I got function code what works for me but I has additional questions:

1. `'message' => 'blablabla'` isn't rendered in DetailView because I add comment via comments widget/tab. 
2. How to force page refresh?

**Reply by mariuszkrzaczkowski** [reply]
It doesn't work for comments, I guess it could be handy to add.
You can try a handler with the EntityBeforeSave type

**Reply by vovpff** [reply]
@mariuszkrzaczkowski  you mean `message` parameter?
What about force page refresh?

**Reply by vovpff** [reply]
> You can try a handler with the EntityBeforeSave type

In this case how can I void new record saving?

**Reply by mariuszkrzaczkowski** [reply]
> > You can try a handler with the EntityBeforeSave type
> 
> In this case how can I void new record saving?

terminate script execution

**Reply by mariuszkrzaczkowski** [reply]
> @mariuszkrzaczkowski you mean `message` parameter? What about force page refresh?

you need to modify the JS files

**Reply by vovpff** [reply]
@mariuszkrzaczkowski thank you. Now I made handler for my needs but I'm stuck with frontend. Need some time. JS difficult for me.

**Reply by mariuszkrzaczkowski** [reply]
Try adding a page reload to the fail function
https://github.com/YetiForceCompany/YetiForceCRM/blob/fba8a4a8a8209db757a68391847868bfd52eff6f/public_html/layouts/basic/modules/Vtiger/resources/Detail.js#L674-L678
